internal/service/transfer/connector_data_source.go

```diff
@@ -0,0 +1,156 @@
+	return &dataSourceConnector{}, nil
+}
+
+const (
+	DSNameConnector = "Connector Data Source"
+)
+
+type dataSourceConnector struct {
+	framework.DataSourceWithConfigure
+}
+
+func (d *dataSourceConnector) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) { // nosemgrep:ci.meta-in-func-name
+	resp.TypeName = "aws_transfer_connector"
+}
+
+func (d *dataSourceConnector) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			// Connector object was expanded
+			"access_role": schema.StringAttribute{
+				Computed: true,
+			},
+			names.AttrARN: schema.StringAttribute{
+				Computed: true,
+			},
+			"as2_config": schema.ListAttribute{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[dsAs2Config](ctx),
+				Computed:   true,
+			},
+			names.AttrID: schema.StringAttribute{
+				CustomType: fwtypes.RegexpType,
+				Required:   true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexache.MustCompile(`c-([0-9a-f]{17})`),
+						""),
+					stringvalidator.LengthAtMost(19),
+					stringvalidator.LengthAtLeast(19),
+				},
+			},
+			"logging_role": schema.StringAttribute{
+				Computed: true,
+			},
+			"security_policy_name": schema.StringAttribute{
+				Computed: true,
+			},
+			"service_managed_egress_ip_addresses": schema.ListAttribute{
+				CustomType: fwtypes.ListOfStringType,
+				Computed:   true,
+			},
+			"sftp_config": schema.ListAttribute{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[dsSftpConfig](ctx),
+				Computed:   true,
+			},
+			"tags": schema.ListAttribute{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[dsTags](ctx),
+				Computed:   true,
+			},
+			"url": schema.StringAttribute{
+				Computed: true,
+			},
+		},
+	}
+}
+
+// TIP: ==== ASSIGN CRUD METHODS ====
+// Data sources only have a read method.
+func (d *dataSourceConnector) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	conn := d.Meta().TransferClient(ctx)
+
+	var data dsConnectorData
+	var describeConnectorInput transfer.DescribeConnectorInput
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if !data.ConnectorId.IsNull() || !data.ConnectorId.IsUnknown() {
+		describeConnectorInput.ConnectorId = data.ConnectorId.ValueStringPointer()
+	}
+
+	description, findConnectorError := conn.DescribeConnector(ctx, &describeConnectorInput)
+
+	if findConnectorError != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.Transfer, create.ErrActionReading, DSNameConnector, data.SecurityPolicyName.String(), findConnectorError),
+			findConnectorError.Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(flex.Flatten(ctx, description.Connector, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
```

Comment:
Since `tags` are handled separately from flatten, they will need to be set

```suggestion
	tags := KeyValueTags(ctx, description.Connector.Tags).IgnoreAWS().IgnoreConfig(d.Meta().IgnoreTagsConfig)
	data.Tags = flex.FlattenFrameworkStringValueMap(ctx, tags.Map())
```